src/jit/morph.cpp

```diff
@@ -7417,6 +7432,45 @@ bool Compiler::fgCanFastTailCall(GenTreeCall* callee)
                     calleeArgRegCount += size;
                 }
 
+#elif defined(_TARGET_ARM_) // ARM
+                var_types hfaType  = GetHfaType(argx);
+                bool      isHfaArg = varTypeIsFloating(hfaType);
+                size_t    size     = 1;
+
+                if (isHfaArg)
+                {
+                    reportFastTailCallDecision("Callee uses float register arguments.", 0, 0);
+                    return false;
+                }
+                else
+                {
+                    size = (unsigned)(roundUp(info.compCompHnd->getClassSize(objClass), TARGET_POINTER_SIZE)) /
+                           TARGET_POINTER_SIZE;
+                    // We cannot handle split struct yet
+                    // TODO: Fix to calculate exact count
+                    if ((calleeArgRegCount < MAX_REG_ARG) && (size + calleeArgRegCount > MAX_REG_ARG))
+                    {
+                        reportFastTailCallDecision("Callee uses split struct argument.", 0, 0);
+                        return false;
+                    }
+
+                    if (size > 1)
+                    {
+                        // hasTwoSlotSizedStruct will determine if the struct value can be passed  multiple slot.
+                        // We set hasTwoSlotSizedStruct if size > 1 because all struct are passed by value on ARM32.
+                        hasTwoSlotSizedStruct = true;
+                        if (calleeArgRegCount >= MAX_REG_ARG)
+                        {
+                            // hasMultiByteStackArgs will determine if the struct can be passed
+                            // in registers. If it cannot we will break the loop and not
+                            // fastTailCall. This is an implementation limitation
+                            // where the callee only is checked for non enregisterable structs.
+                            // It is tracked with https://github.com/dotnet/coreclr/issues/12644.
+                            hasMultiByteStackArgs = true;
```

Comment:
This seems to be using hasMultByteStackArgs in a undesired way as this struct can be passed in registers, there are just more callee arguments and we have to spill. Plus, I am not a big fan of the original hasMultiByteStackArgs code path I would like to avoid building on it.

I think the preferred way of dealing with the case of calleeArgRegCount >= MAX_REG_ARG && hasTwoSlotSizedStruct is here: https://github.com/dotnet/coreclr/blob/master/src/jit/morph.cpp#L7524.